src/control/server/storage/bdev/provider_test.go

```diff
@@ -0,0 +1,307 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+// The Government's rights to use, modify, reproduce, release, perform, display,
+// or disclose this software are subject to the terms of the Apache License as
+// provided in Contract No. 8F-30005.
+// Any reproduction of computer software, computer software documentation, or
+// portions thereof marked with this legend must also reproduce the markings.
+//
+package bdev
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/pkg/errors"
+
+	"github.com/daos-stack/daos/src/control/common"
+	"github.com/daos-stack/daos/src/control/logging"
+	"github.com/daos-stack/daos/src/control/server/storage"
+)
+
+func TestBdevScan(t *testing.T) {
+	for name, tc := range map[string]struct {
+		req    ScanRequest
+		mbc    *MockBackendConfig
+		expRes *ScanResponse
+		expErr error
+	}{
+		"no devices": {
+			req:    ScanRequest{},
+			expRes: &ScanResponse{},
+		},
+		"single device": {
+			req: ScanRequest{},
+			mbc: &MockBackendConfig{
+				ScanRes: storage.NvmeControllers{storage.MockNvmeController()},
+			},
+			expRes: &ScanResponse{
+				Controllers: storage.NvmeControllers{storage.MockNvmeController()},
+			},
+		},
+		"multiple devices": {
+			req: ScanRequest{},
+			mbc: &MockBackendConfig{
+				ScanRes: storage.NvmeControllers{
+					storage.MockNvmeController(1),
+					storage.MockNvmeController(2),
+					storage.MockNvmeController(3),
+				},
+			},
+			expRes: &ScanResponse{
+				Controllers: storage.NvmeControllers{
+					storage.MockNvmeController(1),
+					storage.MockNvmeController(2),
+					storage.MockNvmeController(3),
+				},
+			},
+		},
+		"failure": {
+			req: ScanRequest{},
+			mbc: &MockBackendConfig{
+				ScanErr: errors.New("scan failed"),
+			},
+			expErr: errors.New("scan failed"),
+		},
+	} {
+		t.Run(name, func(t *testing.T) {
+			log, buf := logging.NewTestLogger(name)
+			defer common.ShowBufferOnFailure(t, buf)
+
+			p := NewProvider(log, NewMockBackend(tc.mbc))
+
+			gotRes, gotErr := p.Scan(tc.req)
+			common.CmpErr(t, tc.expErr, gotErr)
+			if gotErr != nil {
+				return
+			}
+
+			if diff := cmp.Diff(tc.expRes, gotRes); diff != "" {
+				t.Fatalf("\nunexpected response (-want, +got):\n%s\n", diff)
+			}
+		})
+	}
+}
+
+func TestBdevPrepare(t *testing.T) {
+	for name, tc := range map[string]struct {
+		req    PrepareRequest
+		mbc    *MockBackendConfig
+		expRes *PrepareResponse
+		expErr error
+	}{
+		"reset fails": {
+			req: PrepareRequest{},
+			mbc: &MockBackendConfig{
+				ResetErr: errors.New("reset failed"),
+			},
+			expErr: errors.New("reset failed"),
+		},
+		"reset-only": {
+			req: PrepareRequest{
+				ResetOnly: true,
+			},
+			expRes: &PrepareResponse{},
+		},
```

Comment:
That's a good suggestion. I'll add it to this one if it needs another go-round, otherwise I'll add it to the follow-on PR that forwards bdev operations to the privileged helper.